src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone A DateTimeZone object representing the desired time zone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format, $default = null, $timeZone = null)
+    {
+        $time = $this->get($key, (string) $default);
```

Comment:
Do it in 2 lines then :) Its more about _not_ casting `$default`, and rely on whatever value `$time` is. (null, valid or invalid).

Hence i would throw an exception when the value is invalid for the _given_ format. Ignoring the user's expected format and just return null can be a real WTF. This is also the difference compared to `getInt`, etc. However it can also be explained as the user responsibility to validate, but imo it would be the added value for this shortcut ;-)

Separating null from void is a good thing :) And this should be accepted by now afaik (https://github.com/symfony/symfony-docs/pull/6614)